tests/core/parser.test.js

```diff
@@ -2,40 +2,16 @@ const parser = require("../../core/parser");
 
 test("Parse tokens", () => {
   const tokens = [
-    { name: "lbra", index: 0, raw: "[" },
-    { name: "number", index: 1, raw: "1" },
-    { name: "number", index: 3, raw: "2" },
-    { name: "number", index: 5, raw: "3" },
-    { name: "rbra", index: 6, raw: "]" },
-    { name: "map", index: 10, raw: "-> *" },
-    { name: "symbol", index: 15, raw: "mul" },
-    { name: "number", index: 19, raw: "2" },
-    { name: "flow_end", raw: "flow_end", index: 20 },
-    { name: "map", index: 23, raw: "-> *" },
-    { name: "symbol", index: 28, raw: "add" },
-    { name: "number", index: 32, raw: "1" },
-    { name: "flow_end", raw: "flow_end", index: 33 },
-    { name: "pipe", index: 36, raw: "->" },
-    { name: "symbol", index: 39, raw: "print" },
-    { name: "flow_end", raw: "flow_end", index: 45 },
-    { name: "lbra", index: 46, raw: "[" },
-    { name: "number", index: 47, raw: "1" },
-    { name: "number", index: 49, raw: "2" },
-    { name: "number", index: 51, raw: "3" },
-    { name: "rbra", index: 52, raw: "]" },
-    { name: "map", index: 56, raw: "-> *" },
-    { name: "symbol", index: 61, raw: "mul" },
-    { name: "number", index: 65, raw: "2" },
-    { name: "flow_end", raw: "flow_end", index: 66 },
-    { name: "map", index: 69, raw: "-> *" },
-    { name: "symbol", index: 74, raw: "add" },
-    { name: "number", index: 78, raw: "1" },
-    { name: "flow_end", raw: "flow_end", index: 79 },
-    { name: "map", index: 82, raw: "-> *" },
-    { name: "symbol", index: 87, raw: "print" },
-    { name: "flow_end", raw: "flow_end", index: 92 },
-    { name: "eof", raw: "eof", index: 92 }
+    { index: 0, name: "string", raw: "'Hello World'" },
+    { index: 14, name: "pipe", raw: "->" },
+    { index: 17, name: "symbol", raw: "print" },
+    { index: 17, name: "eof", raw: "eof" }
   ];
   const [, output] = parser.parse(tokens);
-  expect(output[0].name).toBe("clio");
+  expect(output[0].name).toBe("string");
+});
+
+test("Parse input string", async () => {
+  const output = await parser.parser("'Hello World' -> print\n");
+  expect(output.name).toBe("clio");
```

Comment:
Hmm, why `output.name` is `clio`?

Reply:
I'm not 100% sure, but I think that indicates that the outputted CST is a Clio program. In case it fails, name is undefined. Should've tested that case as well🤔

Reply:
Fixed by ac430c9